server/build_event_protocol/build_event_handler/build_event_handler.go

```diff
@@ -159,14 +159,21 @@ func (e *EventChannel) writeCompletedBlob(ctx context.Context, blobID string, in
 }
 
 func (e *EventChannel) MarkInvocationDisconnected(ctx context.Context, iid string) error {
-	e.statusReporter.ReportDisconnect(ctx)
+	if !e.beValues.FinishedSuccess() {
```

Comment:
I'm a little nervous about this method diverging from FinalizeInvocation. What do you think about a simple early exit at the top:

```
if (e.besValues.Finished()) {
  return e.FinalizeInvocation(iid);
}
```

and deciding to do this whether or not the finish event was received rather than the exit code being successful. Looks like we already look at the exit code in fillInvocationFromEvents:
https://github.com/buildbuddy-io/buildbuddy/blob/f457925906010e188c35e961befb9ecf96134e4d/server/build_event_protocol/event_parser/event_parser.go#L145

Would probably be better to mark an invocation as "failed" if it received the finished event without success rather than disconnected.

Sorry I didn't investigate this well enough when commenting on the issue. Thanks for getting this done so quickly!

Reply:
It looks like a good idea to me. And it's an easy change, no big deal :P